modules/collect/src/main/java/com/opengamma/strata/collect/MapStream.java

```diff
@@ -383,7 +383,7 @@ private MapStream(Stream<Map.Entry<K, V>> underlying) {
    * @param <R>  the type of the new keys
    * @return a stream of entries with new keys from the mapper function assigned to the values
    */
-  public <R> MapStream<R, V> flatMapKeys(Function<? super K, Stream<R>> mapper) {
+  public <R> MapStream<R, V> flatMapKeys(Function<? super K, Stream<? extends R>> mapper) {
```

Comment:
The JDK uses `<R> Stream<R> flatMap(Function<? super T, ? extends Stream<? extends R>> mapper);` (one extra `? extends`)

My concern is whether adding the extra degree of freedom actually makes the job of the compiler harder, such that it might not resolve the desired type at all. Is there a particular use case where it is valuable to have the resulting stream be a supertype of the type of the stream returned from the function?